packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/Mappings/PlanMappingsSection.tsx

```diff
@@ -490,6 +505,24 @@ export const PlanMappingsSection: React.FC<PlanMappingsSectionProps> = ({
     ),
   ];
 
+  const canEditNetworkMapping = (source) => {
+    if (!networkIdsUsedByMigratedVms) {
+      return true;
+    }
+    const idLabelObj = mapSourceNetworksIdsToLabels(sourceNetworks);
+    const id = Object.keys(idLabelObj).find((key) => idLabelObj[key] === source);
+    return !networkIdsUsedByMigratedVms.includes(id);
+  };
+
+  const canEditStorageMapping = (source) => {
+    if (!storageIdsUsedByMigratedVms) {
+      return true;
+    }
+    const idLabelObj = mapSourceStoragesIdsToLabels(sourceStorages);
+    const id = Object.keys(idLabelObj).find((key) => idLabelObj[key] === source);
+    return !storageIdsUsedByMigratedVms.includes(id);
+  };
```

Comment:
consider exporting to a util file

Reply:
I looked into it but would be a bit more involved to extricate it, maybe a good followup